src/components/InfoWindow/Contents/BookMark.tsx

```diff
@@ -1,20 +1,48 @@
-import React from 'react';
 import { toast } from 'react-hot-toast';
 import { BookmarkFilledIcon, BookmarkIcon } from '@assets/index';
 import styled from '@emotion/styled';
+import { MessageToFlutterType } from '@constants/flutterCallback';
+import messageToFlutter from '@libs/webview/messageToFlutter';
+import { useRecoilValue } from 'recoil';
+import { activatedCafeIdAtom } from '@states/infoWindow';
+import { favoriteListAtom } from '@states/favoriteList';
+import { isWebViewAtom } from '@states/isWebView';
+import { useEffect, useRef } from 'react';
 
-interface BookMarkProps {
-  isBookmarked: boolean;
-}
+function BookMark() {
+  const activatedCafeId = useRecoilValue<number | null>(activatedCafeIdAtom);
+  const favoriteList = useRecoilValue(favoriteListAtom);
+  const isBookmarked = activatedCafeId !== null ? favoriteList.includes(Number(activatedCafeId)) : false;
+  const isWebView = useRecoilValue<boolean>(isWebViewAtom);
+  const isMounted = useRef(false);
 
-function BookMark({ isBookmarked }: BookMarkProps) {
   const onBookmarkClick = () => {
-    toast.success(isBookmarked === true ? '북마크가 해제되었습니다.' : '북마크가 추가되었습니다.');
+    if (isWebView) {
```

Comment:
```suggestion
    if (!isWebView) {toast('현재 모바일 환경에서만 북마크를 추가할 수 있습니다.');}
```

로 처리하면 depth를 줄일수 있을거같아요!!